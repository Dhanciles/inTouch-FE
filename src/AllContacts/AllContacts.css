.menu-bar {
  margin-left: 650px;
}

nav {
  display: flex; 
  flex-direction: column; 
  justify-content: space-around; 
  margin-bottom: 1.5em;
}

ul {
  justify-content: space-between; 
  height: 95px; 
  width: 85px;
  padding: 10px; 
  background-color: #5B609F; 
  align-items: center; 
  border-radius: 25px; 
}

li {
  display: flex; 
  flex-direction: row; 
  margin-top: 0.3em;
  font-family:'Open Sans', sans-serif;
  font-size: 14px; 
  font-weight: 800; 
  color: #F2F3F9; 
  list-style: none; 
  cursor: pointer; 
}

.all-contacts-header {
  margin-top: 3em;
  font-family: 'Oswald', sans-serif;
  font-size: 1.5em;
  font-weight: 300;
  color: #191C69;
  text-align: center;
}

.contacts-container {
  margin-top: 1em;
  margin-left: 5%;
  width: 90%;
  height: 150;
  border-radius: 20px;
  border: 2px solid #191C69;
}

.contacts {
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 1em;
  margin-right: 1em;
}

.contact {
  display: flex;
  flex-direction: row; 
  padding: 4px; 
  background-color: #5B609F;
  color: #F2F3F9; 
  border-radius: 5px; 
  margin-bottom: 0.1em;
}

.name-container {
  font-family:'Open Sans', sans-serif;
  font-weight: 900; 
  font-size: 1.1rem; 
  margin-left: 5px;
  margin-right: 20px;
  width: 180px;
  border-right: 1px solid white;
}

.last-contacted-container {
  font-family:'Open Sans', sans-serif;
  font-weight: 900; 
  font-size: 1.1rem;
  color: #F2F3F9; 
  margin-left: 180px; 
}
