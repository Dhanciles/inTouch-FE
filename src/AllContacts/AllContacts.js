import React, { Component } from 'react'; 
import './AllContacts.css'; 

export default class AllContacts extends Component {
  constructor() {
    super()
  }
  
  render() {
    let allContacts = [{name: 'Derek', lastContacted: '2019-01-17'}, {name: 'Rajaa', lastContacted: '2019-01-18'}, {name: 'Will', lastContacted: '2019-01-19'}]
    let renderedContacts = allContacts.map(contact => {
      return (
        <div className="contact"> 
          <p className="contact-name">{contact.name}</p>
          <p className="last-contacted">Last Contacted: {contact.lastContacted}</p>
        </div>
      )
    })
    return(
      <div className="all-contacts">
        <div className="all-contacts-header">Contacts</div>
        <div className="contacts-container">
          <div className="contacts"> {renderedContacts} </div>
        </div>
      </div>
    )
  }
}