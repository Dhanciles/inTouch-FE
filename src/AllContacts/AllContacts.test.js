import React from 'react'; 
import { shallow } from 'enzyme'; 
import AllContacts from './AllContacts.js'; 

describe('AllContacts', () => {
  let allContacts;

  beforeEach(() => {
    allContacts =[{name: 'Derek', lastContacted: '2019-01-17'}, {name: 'Rajaa', lastContacted: '2019-01-18'}, {name: 'Will', lastContacted: '2019-01-19'}]; 
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<AllContacts />)

    expect(wrapper).toMatchSnapshot()
  });
});