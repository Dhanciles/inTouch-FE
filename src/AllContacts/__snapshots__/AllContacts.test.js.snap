// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllContacts should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllContacts />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="all-contacts-header"
        >
          Contacts
        </div>,
        <div
          className="contacts-container"
        >
          <div
            className="contacts"
          >
             
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Derek
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-17
              </p>
            </div>
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Rajaa
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-18
              </p>
            </div>
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Will
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-19
              </p>
            </div>
             
          </div>
        </div>,
      ],
      "className": "all-contacts",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Contacts",
          "className": "all-contacts-header",
        },
        "ref": null,
        "rendered": "Contacts",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="contacts"
          >
             
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Derek
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-17
              </p>
            </div>
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Rajaa
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-18
              </p>
            </div>
            <div
              className="contact"
            >
              <p
                className="contact-name"
              >
                Will
              </p>
              <p
                className="last-contacted"
              >
                Last Contacted: 
                2019-01-19
              </p>
            </div>
             
          </div>,
          "className": "contacts-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              Array [
                <div
                  className="contact"
                >
                  <p
                    className="contact-name"
                  >
                    Derek
                  </p>
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-17
                  </p>
                </div>,
                <div
                  className="contact"
                >
                  <p
                    className="contact-name"
                  >
                    Rajaa
                  </p>
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-18
                  </p>
                </div>,
                <div
                  className="contact"
                >
                  <p
                    className="contact-name"
                  >
                    Will
                  </p>
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-19
                  </p>
                </div>,
              ],
              " ",
            ],
            "className": "contacts",
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="contact-name"
                  >
                    Derek
                  </p>,
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-17
                  </p>,
                ],
                "className": "contact",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Derek",
                    "className": "contact-name",
                  },
                  "ref": null,
                  "rendered": "Derek",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Last Contacted: ",
                      "2019-01-17",
                    ],
                    "className": "last-contacted",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Last Contacted: ",
                    "2019-01-17",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="contact-name"
                  >
                    Rajaa
                  </p>,
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-18
                  </p>,
                ],
                "className": "contact",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Rajaa",
                    "className": "contact-name",
                  },
                  "ref": null,
                  "rendered": "Rajaa",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Last Contacted: ",
                      "2019-01-18",
                    ],
                    "className": "last-contacted",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Last Contacted: ",
                    "2019-01-18",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="contact-name"
                  >
                    Will
                  </p>,
                  <p
                    className="last-contacted"
                  >
                    Last Contacted: 
                    2019-01-19
                  </p>,
                ],
                "className": "contact",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Will",
                    "className": "contact-name",
                  },
                  "ref": null,
                  "rendered": "Will",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Last Contacted: ",
                      "2019-01-19",
                    ],
                    "className": "last-contacted",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Last Contacted: ",
                    "2019-01-19",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            " ",
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="all-contacts-header"
          >
            Contacts
          </div>,
          <div
            className="contacts-container"
          >
            <div
              className="contacts"
            >
               
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Derek
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-17
                </p>
              </div>
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Rajaa
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-18
                </p>
              </div>
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Will
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-19
                </p>
              </div>
               
            </div>
          </div>,
        ],
        "className": "all-contacts",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Contacts",
            "className": "all-contacts-header",
          },
          "ref": null,
          "rendered": "Contacts",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="contacts"
            >
               
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Derek
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-17
                </p>
              </div>
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Rajaa
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-18
                </p>
              </div>
              <div
                className="contact"
              >
                <p
                  className="contact-name"
                >
                  Will
                </p>
                <p
                  className="last-contacted"
                >
                  Last Contacted: 
                  2019-01-19
                </p>
              </div>
               
            </div>,
            "className": "contacts-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                Array [
                  <div
                    className="contact"
                  >
                    <p
                      className="contact-name"
                    >
                      Derek
                    </p>
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-17
                    </p>
                  </div>,
                  <div
                    className="contact"
                  >
                    <p
                      className="contact-name"
                    >
                      Rajaa
                    </p>
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-18
                    </p>
                  </div>,
                  <div
                    className="contact"
                  >
                    <p
                      className="contact-name"
                    >
                      Will
                    </p>
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-19
                    </p>
                  </div>,
                ],
                " ",
              ],
              "className": "contacts",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="contact-name"
                    >
                      Derek
                    </p>,
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-17
                    </p>,
                  ],
                  "className": "contact",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Derek",
                      "className": "contact-name",
                    },
                    "ref": null,
                    "rendered": "Derek",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Last Contacted: ",
                        "2019-01-17",
                      ],
                      "className": "last-contacted",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Last Contacted: ",
                      "2019-01-17",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="contact-name"
                    >
                      Rajaa
                    </p>,
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-18
                    </p>,
                  ],
                  "className": "contact",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Rajaa",
                      "className": "contact-name",
                    },
                    "ref": null,
                    "rendered": "Rajaa",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Last Contacted: ",
                        "2019-01-18",
                      ],
                      "className": "last-contacted",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Last Contacted: ",
                      "2019-01-18",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="contact-name"
                    >
                      Will
                    </p>,
                    <p
                      className="last-contacted"
                    >
                      Last Contacted: 
                      2019-01-19
                    </p>,
                  ],
                  "className": "contact",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Will",
                      "className": "contact-name",
                    },
                    "ref": null,
                    "rendered": "Will",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Last Contacted: ",
                        "2019-01-19",
                      ],
                      "className": "last-contacted",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Last Contacted: ",
                      "2019-01-19",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              " ",
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
